### DISCUSSION TITLE
how to test if script is actually in debug mode (to disable timeouts)

### DISCUSSION THREAD

**Discussion by ugur-kurnaz** [ruby/debug#1011] [Q&A]
Hello,

Is there any instruction that could be used by script to check if debug is enabled? for example, one would like to add timeout to running code. But when debugging those timeouts are a bit troublesome. Rather than modifying timeouts in hard (that may be calculated during runtime), is there a way to check that there is a debugger running like :
`

class Job

  MIN_TIMEOUT = 10
  MAX_TIMEOUT = 30

  def initialize
    @timeout = <is_debug_enabled> ? nil : MAX_TIMEOUT
    @memory = nil
    @result = nil
    @error = []
  end

end
`

thank you for your help

**Comment by st0012** [comment]
I usually use this: `defined?(DEBUGGER__::SESSION)`. When the debugger is active it returns `"constant"`. Otherwise it returns `nil`.

**Comment by ko1** [comment]
`defined?(debugger)` is another way if you don't define `debugger` method.
